src/interpolation_fe.jl

```diff
@@ -0,0 +1,176 @@
+
+using Random
+pt = Point(rand(2))
+pts = [Point(rand(2)) for i in 1:3]
+
+# The finite element function $f_h$ can be evaluated at arbitrary points (or
+# array of points) by
+
+fₕ(pt), fₕ.(pts)
+
+# We can also check our results using
+
+@test fₕ(pt) ≈ f₁(pt)
+@test fₕ.(pts) ≈ f₁.(pts)
+
+# Now let us define the new triangulation $\mathcal{T}_2$ of
+# $\Omega$. We define the map
+
+function sinusoidal(p::Point)
+  r, s = p
+  x = r + 0.05*sin(2π*r)*sin(2π*s)
+  y = s + 0.05*sin(2π*r)*sin(2π*s)
+  Point(x,y)
+end
+
+# to build the new triangulation using a partition of 20 cells per
+# direction. The map can be passed as an argument to
+# `CartesianDiscreteModel` to define the position of the vertices in
+# the new mesh.
+
+partition = (20,20)
+𝒯₂ = CartesianDiscreteModel(domain,partition; map=sinusoidal)
+
+# As before, we define the new `FESpace` consisting of second order
+# elements
+
+reffe₂ = ReferenceFE(lagrangian, Float64, 2)
+V₂ = FESpace(𝒯₂, reffe₂)
+
+# Now we interpolate $f_h$ onto $V_2$ to obtain the new function
+# $g_h$. The first step is to create the `Interpolable` version of
+# $f_h$.
+
+ifₕ = Interpolable(fₕ)
+
+# Then to obtain $g_h$, we dispatch `ifₕ` and the new `FESpace` $V_2$
+# to the `interpolate_everywhere` method of `Gridap`.
+
+gₕ = interpolate_everywhere(ifₕ, V₂)
+
+# Like earlier we can check our results
+
+@test fₕ(pt) ≈ gₕ(pt) ≈ f₁(pt)
+@test fₕ.(pts) ≈ gₕ.(pts) ≈ f₁.(pts)
+
+# We can visualize the results using Paraview
+
+writevtk(get_triangulation(fₕ), "source", cellfields=["fₕ"=>fₕ])
+writevtk(get_triangulation(gₕ), "target", cellfields=["gₕ"=>gₕ])
+
+# which produces the following output
+
+# ![Target](../assets/interpolation/target.png)
+
+# ## Interpolating between Raviart-Thomas FESpaces
+
+# The procedure is identical to Lagrangian finite element spaces, as
+# discussed in the previous section. The extra thing here is that
+# functions in Raviart-Thomas spaces are vector-valued. The degrees of
+# freedom of the RT spaces are fluxes of the function across the edge
+# of the element. Refer to the
+# [tutorial](https://gridap.github.io/Tutorials/dev/pages/t007_darcy/)
+# on Darcy equation with RT for more information on the RT
+# elements.
+
+# Assuming a function
+
+f₂(x) = VectorValue([x[1], x[2]])
+
+# on the domain, we build the associated finite dimensional version
+# $f_h \in V_1$.
+
+reffe₁ = ReferenceFE(raviart_thomas, Float64, 1) # RT space of order 1
+V₁ = FESpace(𝒯₁, reffe₁)
+fₕ = interpolate_everywhere(f₂, V₁)
+
+# As before, we can evaluate the RT function on any arbitrary point in
+# the domain.
+
+fₕ(pt), fₕ.(pts)
+
+# Constructing the target RT space and building the `Interpolable`
+# object,
+
+reffe₂ = ReferenceFE(raviart_thomas, Float64, 2) # RT space of order 2
+V₂ = FESpace(𝒯₂, reffe₂)
+ifₕ = Interpolable(fₕ)
+
+# we can construct the new `FEFunction` $g_h \in V_2$ from $f_h$
+
+gₕ = interpolate_everywhere(ifₕ, V₂)
+
+# Like earlier we can check our results
+
+gₕ(pt), f₂(pt)
```

Comment:
I assume you removed the tests because they are failing... I would change the problem such that the tests pass, instead of removing the tests.

Reply:
You can try just moving between linear spaces (i.e. linear V_2), you have already used `order = 2` in Lagrangian.